### DISCUSSION TITLE
TuYa TS0601_dimmer_3 - Brigtness setting controlled via Home assistant do not work

### DISCUSSION THREAD

**Discussion by swinster** [Koenkk/zigbee2mqtt#18084] [Devices]
Hi,

I have a weird issue with this dimmer (https://www.zigbee2mqtt.io/devices/TS0601_dimmer_3.html). The brightness settings on each gang are between 0-254; however, when controlling the brightness from within Homeassitant, the switch doesn't change the brightness of the bulbs on the circuit. The brightness slider in HA is between 0-100%, yet the bit value does get sent to Z2M as I see the settings update; the logs show an `MQTT publish` message with a payload of the equivalent bit value of the brightness percentage, and the status shown in Z2M shows the bit value set.  However, the light brightness never changes. 

If I control the brightness value directly from the "exposes" section of the device page within Z2M, the brightness DOES update, _but_ only after the dimmer gang has been toggled off/on. 

Both HA and Z2M values are updated when a change is made in the opposing app. 

TBH, I don't know if this is a Z2M or a HA issue, but given that things seem to be updated correctly in Z2M from changes in HA, then it feels more like a Z2M problem. If anyone has any insight, I would be interested to know so I can fix the issue.

Z2M version: [1.30.3](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.30.3) commit: [24c6b2e](https://github.com/Koenkk/zigbee2mqtt/commit/24c6b2e)

HA Version: 2023.1.1 (docker version)

Cheers

**Comment by swinster** [comment]
FWIW, I re-pulled the Z2M container so upgrade to [1.31.2](https://github.com/Koenkk/zigbee2mqtt/releases/tag/1.31.2) commit: [21f5125](https://github.com/Koenkk/zigbee2mqtt/commit/21f5125), and, if anything, it seems to have made matters worse. Once the device has been attempted to be controlled in HA, it stops responding. I then seem to have to wait for a while (say 1 minute) to regain control via the exposes tab in Z2M.

Indeed, after a short while, I lose control via Z2M as well, and then I struggle to regain control.

**Comment by swinster** [comment]
This is driving me nuts. I am now thinking it might be the dimmer itself. If you manually press the controls on the dimmer, you can see the brightness sliders in the Exposes tab of the device in Z2M move; however, the control in the opposite direction (From Z2M --> Device) is sketchy. Sometimes it works, sometimes it doesn't, sometimes you get an error, and sometimes you don't.

Is there anything I can do to troubleshoot this further?

**Comment by swinster** [comment]
FWIW, I even requested a replacement dimmer from Moes. The has just arrived and has been wired in, and initially, all seems a lot better. The gang would not dim (either via the hardware button or Z2M) now dims, and everything works as expected from the dimmer itself. 

However, Z2M seems to have an issue on occasion. I have found that refreshing the "Exposes" pages seem to help, but this defiantly seems like a Z2M issue. I will continue to test, and if there is a continued problem, I will escalate this to an issue.

**Comment by coph** [comment]
I have the same issue. I trying to change the light_type without success.

**Comment by coph** [comment]
Just for your information... I've re-paired the device in HA and it started to work.

**Reply by coph** [reply]
... for some time. But get the issue again.

**Reply by Frankpiet** [reply]
Al een oplossing?  
\

**Comment by coph** [comment]
Sorry I do not speak german.

Paul

Le lun. 28 avr. 2025 à 10:23, Frankpiet ***@***.***> a écrit :

> Al een oplossing?
> \
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/Koenkk/zigbee2mqtt/discussions/18084#discussioncomment-12970650>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AFMRWSXFVX62WG2BUJ6U2PD23Y2WLAVCNFSM6AAAAAB4AR5XXKVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTEOJXGA3DKMA>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


--
